### PR TITLE
build-rust-crate: add missing CARGO_PKG env variables

### PR DESCRIPTION
**Pull request by bouk**
## Description of changes

Some crate builds might fail because not all the [environment variables](https://doc.rust-lang.org/cargo/reference/environment-variables.html) that cargo exposes are used in build-rust-crate. This adds the missing ones that I could see.

## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [x] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by bouk** [comment]
@RaitoBezarius could you take a look at this one?

**Review by RaitoBezarius — CHANGES_REQUESTED** [review]
Please put the source for those environment variables in the source code and in the commit message.

**Comment by bouk** [comment]
@RaitoBezarius done!

**Review by RaitoBezarius — APPROVED** [review]

File: pkgs/build-support/rust/build-rust-crate/configure-crate.nix [hunk 1]
```diff
@@ -147,6 +154,11 @@ in ''
   export CARGO_PKG_VERSION_PATCH=${lib.elemAt version 2}
   export CARGO_PKG_VERSION_PRE="${versionPre}"
   export CARGO_PKG_HOMEPAGE="${crateHomepage}"
+  export CARGO_PKG_LICENSE="${crateLicense}"
+  export CARGO_PKG_LICENSE_FILE="${crateLicenseFile}"
+  export CARGO_PKG_README="${crateReadme}"
+  export CARGO_PKG_REPOSITORY="${crateRepository}"
+  export CARGO_PKG_RUST_VERSION="${crateRustVersion}"
```
Comment (apoelstra) [inline pkgs/build-support/rust/build-rust-crate/configure-crate.nix:161]: This unconditionally overrides the environment variable with an empty string in the case that `crate.rust-version` is unset. This is not set anywhere in nixpkgs or in crate2nix. Therefore this just clobbers the environment variable, breaking code that used to work before this change.

Comment (apoelstra) [reply pkgs/build-support/rust/build-rust-crate/configure-crate.nix:161]: Even if I weren't 9 months late on this, I think this is crate2nix's problem, not buildRustCrate's problem.

Users of crate2nix can override this, assuming they know their MSRV and have it in a `msrv` variable, by doing

```nix
buildRustCrateForPkgs = pkgs: crate: pkgs.buildRustCrate (crate // { rust-version = msrv; })
```

which is a little ugly but arguably cleaner than the previous workaround (using `crate.override.preUnpack` to set the environment variables directly).

**Pull request merged**